src/plugins/intel_gna/src/gna_plugin.cpp

```diff
@@ -1778,11 +1801,26 @@ std::vector<std::shared_ptr<const ov::Node>> GNAPlugin::GetOutputs() {
 }
 
 InferenceEngine::QueryNetworkResult GNAPlugin::QueryNetwork(const InferenceEngine::CNNNetwork& network,
-                                                            const std::map<std::string, std::string>& config) const {
+                                                            const std::map<std::string, std::string>& config_map) const {
     InferenceEngine::QueryNetworkResult res;
 
-    if (network.getFunction()) {
-        IE_THROW(NotImplemented) << " ngraph::Function is not supported natively";
+    GNAPluginNS::Config qn_config = config;
+    qn_config.UpdateFromMap(config_map);
+
+    auto effectiveGnaCompileTargetValue = effectiveGnaCompileTarget();
+    auto model = network.getFunction();
+    if (model) {
+        auto supported = GetSupportedNodes(model,
+        [&](std::shared_ptr<ov::Model>& model) {
+                TransformModel(model, effectiveGnaCompileTargetValue, qn_config);
+            },
+        [&](const std::shared_ptr<ngraph::Node>& op) {
+            return ov::intel_gna::ngraph_util::is_op_supported(op, effectiveGnaCompileTargetValue, qn_config);
```

Comment:
using namespace ov::intel_gna::ngraph_util maybe?

Reply:
> using namespace ov::intel_gna::ngraph_util maybe?

I would like to keep it as is. It is used in 2 places only.